oximeter/oxql-types/src/query_summary.rs

```diff
@@ -0,0 +1,49 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Types representing summaries of queries against the timeseries database.
+
+// Copyright 2024 Oxide Computer Company
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use uuid::Uuid;
+
+/// A count of bytes / rows accessed during a query.
+#[derive(Clone, Copy, Debug, Deserialize, JsonSchema, Serialize)]
+pub struct IoCount {
+    /// The number of bytes accessed.
+    pub bytes: u64,
+    /// The number of rows accessed.
+    pub rows: u64,
+}
+
+impl std::fmt::Display for IoCount {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{} rows ({} bytes)", self.rows, self.bytes)
+    }
+}
+
+/// Summary of the I/O resources used by a query.
+#[derive(Clone, Copy, Debug, Deserialize, JsonSchema, Serialize)]
+pub struct IoSummary {
+    /// The bytes and rows read by the query.
+    pub read: IoCount,
+    /// The bytes and rows written by the query.
+    pub written: IoCount,
+}
```

Comment:
If you look at the generated OpenAPI, you can see we end up with no indication on these type names or their doc comments that these have to do with OxQL. Maybe just change "during a query" to "during an OxQL query"? Putting it in the names might be overkill. Another thing you could do is keep these as-is here and implement trivial analogues in `views.rs` that are the same but with slightly more user-friendly names. It's not a huge deal because you can easily tell in the generated code that `IoCount` and `IoSummary` are only used on `QuerySummary` and `QuerySummary` is only used in `OxqlQueryResult`.